manufacturing-client/src/main.rs

```diff
@@ -295,11 +298,70 @@ async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType>
         return Ok(CborSimpleType::Text(mfg_info.into_string().unwrap()));
     }
     match mfg_string_type {
-        MfgStringType::SerialNumber => {
-            let serial = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+        MfgStringType::SerialNumberExt => {
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
                 .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial))
+            let mut arg: String;
+            let output_ifaces = Command::new("ls")
+                .arg("/sys/class/net/")
+                .stdout(std::process::Stdio::piped())
+                .stderr(std::process::Stdio::piped())
+                .output()
+                .expect("Failed to execute command");
+
```

Comment:
Probably want to
```rust
   let st = output_ifaces.status;
   if !st.success() {
      return anyhow::bail!("Failed to execute ls: {st:?}");
   }
```

Reply:
(Also now that I think about it, instead of forking `ls` it's likely better to use https://doc.rust-lang.org/std/fs/fn.read_dir.html )

Reply:
Thanks for the suggestions , taken care in the latest commit !

Reply:
Looks like it is still forking and not using `fs::read_dir`?

Reply:
I do agree, please use `fs:read_dir`

Reply:
done ,thanks for suggesting !